plugins/org.eclipse.n4js.ts/src/org/eclipse/n4js/ts/TypeExpressions.xtext

```diff
@@ -334,9 +338,13 @@ N4Keyword:
 	| 'out'
 ;
 
+
 terminal IDENTIFIER:
 	IDENTIFIER_START IDENTIFIER_PART*;
 
+terminal IDENTIFIER_WITH_DASH:
+	IDENTIFIER_START IDENTIFIER_PART+ ('-' IDENTIFIER_PART+)+;
```

Comment:
This will probably break lexing of binary expressions of the shape `abc-efg`

Reply:
I'd rather make this a datatype rule and validate the absence of whitespace in the value converter.

Reply:
Yes, it doesn't work right. Also, the second character should also be allowed to be a '-'. Actually, I would like to wait and get this done with your help later. ;-)

Reply:
Okay, I just read your second comment. Thanks, maybe I try that!